homeassistant/components/ryse/cover.py

```diff
@@ -0,0 +1,129 @@
+"""Support for RYSE Smart Shades via BLE."""
+
+import logging
+from typing import Any
+
+from ryseble.device import RyseBLEDevice
+from ryseble.packets import build_get_position_packet, build_position_packet
+
+from homeassistant.components.cover import CoverEntity, CoverEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import CONF_ADDRESS, CONF_RX_UUID, CONF_TX_UUID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up RYSE Smart Shade cover from a config entry."""
+    device = RyseBLEDevice(
+        entry.data[CONF_ADDRESS],
+        entry.data[CONF_RX_UUID],
+        entry.data[CONF_TX_UUID],
+    )
+    async_add_entities([SmartShadeCover(device)])
+
+
+class SmartShadeCover(CoverEntity):
+    """Representation of a RYSE Smart Shade BLE cover entity."""
+
+    def __init__(self, device: RyseBLEDevice) -> None:
+        """Initialize the Smart Shade cover entity."""
+        self._device = device
+        self._attr_name = f"Smart Shade {device.address}"
```

Comment:
You must set `has_entity_name` and as this is the main feature of the device, you can set the name to None.
```suggestion
    _attr_has_entity_name = True
    _attr_name = None

    def __init__(self, device: RyseBLEDevice) -> None:
        """Initialize the Smart Shade cover entity."""
        self._device = device
```